var/spack/repos/builtin/packages/slate/package.py

```diff
@@ -165,25 +165,31 @@ def mpi_launcher(self):
         commands = ["srun", "mpirun", "mpiexec"]
         return which(*commands, path=searchpath) or which(*commands)
 
-    def test(self):
-        if self.spec.satisfies("@2020.10.00") or "+mpi" not in self.spec:
-            print("Skipping: stand-alone tests")
-            return
+    def test_slate(self):
+        """Test slate with +mpi and other options"""
+
+        if self.spec.satisfies("@2020.10.00") and "+mpi" not in self.spec:
+            raise SkipTest(
+                "Package must be installed with +mpi and as version @2021.05.01 or later"
+            )
+        elif self.spec.satisfies("@2020.10.00"):
+            raise SkipTest("Package must be installed as version @2021.05.01 or later")
+        elif "+mpi" not in self.spec:
+            raise SkipTest("Package must be installed with +mpi")
```

Comment:
There's too much here, especially since the PR to deprecate `+mpi` is going to be changing this check anyway.  Better to keep the original.